index.js

```diff
@@ -795,15 +813,13 @@
   //. ['oo!', 'oo?', 'on!', 'on?', 'no!', 'no?', 'nn!', 'nn?']
   //.
   //. > chainRec(
-  //. .   [],
+  //. .   Array,
   //. .   (next, done, x) => Array.isArray(x) ? x.map(next) : [done(x)],
   //. .   [1, [[2, 3], 4], 5]
   //. . )
   //. [1, 2, 3, 4, 5]
   //. ```
-  var chainRec = function(chainRec, f, x) {
-    return ChainRec.getBoundMethod('chainRec', chainRec)(f, x);
-  };
+  var chainRec = ChainRec.staticMethods.chainRec;
 
   //# filter :: (Monad m, Monoid m a) => (a -> Boolean, m a) -> m a
```

Comment:
Actually you are write! We can't `reduce`: `Task`, `IO` or `Function`. But there are structures which can't be a Monad but are Foldable: Comonads (i.e. CoFree) and ZipList and maybe others. What if we have two filter functions like this maybe: `filter` and `filterM`